src/services/EdFi.DataManagementService.Api/Core/ICoreFacade.cs

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+using EdFi.DataManagementService.Api.Core.Model;
+
+namespace EdFi.DataManagementService.Api.Core;
+
+/// <summary>
+/// The facade a frontend uses to access DMS Core.
+///
+/// The intent of this design is to provide a web framework-independent interface
+/// for 1) ease of testing and 2) ease of supporting future frontends e.g.
+/// AWS Lambda, Azure functions, etc.
+/// </summary>
+public interface ICoreFacade
+{
+    /// <summary>
+    /// DMS entry point for API upsert requests
+    /// </summary>
+    public Task<FrontendResponse> Upsert(FrontendRequest frontendRequest);
```

Comment:
Consider renaming these with "Async" to follow common C# idiom, e.g. `UpsertAsync`.

Since there is no alternative, I'm of an open mind on this point.